### ISSUE TITLE
dependency() - main keyword argument gives warning

### ISSUE THREAD

**Issue by afg984** [mesonbuild/meson#2828]
Running meson with https://github.com/mesonbuild/meson/tree/master/test%20cases/frameworks/2%20gtest gives: 
```
The Meson build system
Version: 0.44.0
Source dir: /home/afg/Downloads/meson/test cases/frameworks/2 gtest
Build dir: /home/afg/Downloads/meson/test cases/frameworks/2 gtest/build
Build type: native build
Project name: gtest
Native C++ compiler: c++ (gcc 7.2.1)
Build machine cpu family: x86_64
Build machine cpu: x86_64
WARNING: Passed invalid keyword argument "main" in meson.build line 3.
This will become a hard error in the future.
Dependency GTest found: YES (prebuilt)
WARNING: Passed invalid keyword argument "main" in meson.build line 4.
This will become a hard error in the future.
Dependency GTest found: YES (prebuilt)
Build targets in project: 2
Found ninja-1.8.2 at /usr/bin/ninja
```
Despite the warning, the `main` argument does affect wheter `-lgtest_main` is present on the compiling command.

Probably related: #665

**Comment by W4RH4WK** [comment]
I am currently using the latest *gtest wrap* and am doing the following:

```meson
gtest = dependency('gtest', fallback: ['gtest', 'gtest_main_dep'])
```

If I need the *mainless* version of gtest I'd be passing `gtest_dep` as second argument to `fallback` instead.

**Comment by nirbheek** [comment]
Fixed in #2828.
